src/MoBi.Core/Serialization/Services/SerializationContextFactory.cs

```diff
@@ -49,9 +51,13 @@ public SerializationContext Create(SerializationContext parentSerializationConte
             allRepositories.AddRange(parentSerializationContext.Repositories);
             parentSerializationContext.IdRepository.All().Each(idRepository.Register);
          }
-         
+
+         //We only registers existing simulation if we are not deserializing a whole Simulation.
+         //Otherwise we may get conflicts when loading an existing simulation again
+         var shouldRegisterSimulations = type != typeof(SimulationTransfer);
```

Comment:
This is very interesting. MoBi does not use the service provided by core to load a simulation transfer (WHY!?!?) 
So we need to make sure that we do not register OTHER simulations when loading a simulation (this is a very strange use case but eh..it should not crash)

Reply:
Hmm. So now all objects of a loaded simulation are not registered at all?

Reply:
No that would be a problem.
A simulation is not registered during deserializatioj in the temp repository (created with new a few lines before) when one deserializes another simulation. (They should not know of each other anyways, this is just for object references)